proto/rill/local/v1/api.proto

```diff
@@ -42,3 +51,56 @@ message GetVersionResponse {
   string current = 1;
   string latest = 2;
 }
+
+message DeployValidationRequest {}
+
+message DeployValidationResponse {
+  bool is_authenticated = 1; // if true below fields are relevant after login
+  bool is_github_connected = 2; // if true below fields are relevant after git install
+  string git_grant_access_url = 3; // redirect to this if is_github_connected or is_repo_access_granted is false
+  string git_user_name = 4;
+  bool is_user_app_installed = 5; // if rill git app is installed on user account
+  string user_app_permission = 6;
+  repeated string git_user_orgs = 7;
+  bool is_git_repo = 8;
+  bool git_remote_found = 9; // only applicable when is_git_repo is true
+  bool is_repo_access_granted = 10; // relevant only when is_git_repo is true and remote found, if false redirect to git_grant_access_url
```

Comment:
Since it's github-specific, put `github` in the name. Also consider moving it next to the Github-related properties